Microsoft.TeamServices.Samples.Client/Release/ReleasesSample.cs

```diff
@@ -0,0 +1,512 @@
+
+            return releaseDefinition;
+        }
+
+        [ClientSampleMethod]
+
+        public ReleaseDefinition CreateAReleaseDefinition()
+        {
+            string projectName = ClientSampleHelpers.FindAnyProject(this.Context).Name;
+
+            // Get a release client instance
+            VssConnection connection = Context.Connection;
+            ReleaseHttpClient releaseClient = connection.GetClient<ReleaseHttpClient>();
+
+            // JSON to create release definition
+            string releaseDefinitionJson = "{\"name\":\"Fabrikam-web\",\"revision\" : 1,\"environments\":[{\"name\":\"PROD\",\"deployPhases\":[{\"name\":\"Run on agent\",\"phaseType\":1,\"rank\":1, \"deploymentInput\":{\"queueId\":2}}],\"preDeployApprovals\":{\"approvals\":[{\"rank\":1,\"isAutomated\":true}]},\"postDeployApprovals\":{\"approvals\":[{\"rank\":1,\"isAutomated\":true}]},\"retentionPolicy\":{\"daysToKeep\":30,\"releasesToKeep\":3,\"retainBuild\":true}}]}";
+
+            ReleaseDefinition definition = JsonConvert.DeserializeObject<ReleaseDefinition>(releaseDefinitionJson);
+
+            // create a release definition
+            ReleaseDefinition releaseDefinition = releaseClient.CreateReleaseDefinitionAsync(project: projectName, releaseDefinition: definition).Result;
+
+            Console.WriteLine("{0} {1}", releaseDefinition.Id.ToString().PadLeft(6), releaseDefinition.Name);
+
+            return releaseDefinition;
+        }
+
+        [ClientSampleMethod]
+
+        public ReleaseDefinition UpdateAReleaseDefinition()
+        {
+            string projectName = ClientSampleHelpers.FindAnyProject(this.Context).Name;
+
+            // Get a release client instance
+            VssConnection connection = Context.Connection;
+            ReleaseHttpClient releaseClient = connection.GetClient<ReleaseHttpClient>();
+
+            List<ReleaseDefinition> releaseDefinitions = releaseClient.GetReleaseDefinitionsAsync(project: projectName, searchText: releaseDefinitionName).Result;
+            int releaseDefinitionId = releaseDefinitions.FirstOrDefault().Id;
+            ReleaseDefinition releaseDefinition = releaseClient.GetReleaseDefinitionAsync(project: projectName, definitionId: releaseDefinitionId).Result;
+
+            // add a non secret variable to definition
+            ConfigurationVariableValue variable = new ConfigurationVariableValue();
+            variable.Value = "NonSecretValue";
+            variable.IsSecret = false;
+            releaseDefinition.Variables.Add("NonSecretVariable", variable);
+
+            // update release definition
+            ReleaseDefinition updatedReleaseDefinition = releaseClient.UpdateReleaseDefinitionAsync(project: projectName, releaseDefinition: releaseDefinition).Result;
+
+            Console.WriteLine("{0} {1}", updatedReleaseDefinition.Id.ToString().PadLeft(6), updatedReleaseDefinition.Name);
```

Comment:
Is there a "revision" or "last modifed date" field that you could show here? This would demonstrate the update worked.